tests/models/test_gpu.py

```diff
@@ -190,10 +191,12 @@ def test_torchelastic_gpu_parsing(mocked_device_count, mocked_is_available, gpus
     sanitizing the gpus as only one of the GPUs is visible."""
     trainer = Trainer(gpus=gpus)
     assert isinstance(trainer._accelerator_connector.cluster_environment, TorchElasticEnvironment)
-    assert trainer.gpus == gpus
```

Comment:
Why not keep the assert with `num_devices`?

Reply:
Could add that.